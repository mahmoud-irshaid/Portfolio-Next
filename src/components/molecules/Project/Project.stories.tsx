/* eslint-disable import/no-anonymous-default-export */
import { ProjectProps } from './Project';
import { Project } from './Project';

import { mockProject } from '@mockdata';

export default {
	title: 'molecules/Project',
	component: Project,
};

export const ProjectStory = (args: ProjectProps) => <Project {...args} />;

ProjectStory.storyName = 'Project';
ProjectStory.args = {
	...mockProject,
	index: 0,
};
