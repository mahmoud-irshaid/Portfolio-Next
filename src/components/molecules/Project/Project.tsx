import { Box } from '@components/atoms/Box';
import { FloatingImages } from '@components/atoms/FloatingImages';

import type { Project as ProjectType } from '@types';
import classNames from 'classnames';
import Image from 'next/image';
import Link from 'next/link';

export interface ProjectProps extends ProjectType {
	index: number;
}

const Project = ({
	slug,
	title,
	skills,
	mainImage,
	secondaryImages,
	index,
	description
}: ProjectProps) => {
	const isEven = index % 2 === 0;

	return (
		<article className="mt-24 transition first-of-type:mt-0 md:hover:scale-[1.01] lg:w-4/5">
			<Link href={slug ? slug : '#'} target="_blank" className="group">
				<Box>
					<div
						className={classNames('flex flex-col gap-4', {
							'md:flex-row': isEven,
							'md:flex-row-reverse': !isEven,
						})}
					>
						<div className="flex h-full flex-col md:w-1/2">
							<h1 className="mb-2 text-xl font-bold md:text-3xl">{title}</h1>
							<p className="max-w-xs text-sm">
								<strong>Tech used: </strong>
								{skills.join(', ')}
							</p>
							<p className="max-w-xs text-sm mt-4">{description}</p>
						</div>
						<div
							className={classNames(
								'-mx-10 -mb-12 md:mx-0 md:-mb-12 md:-mt-8',
								{
									'md:-mr-20': isEven,
									'md:-ml-20': !isEven,
								}
							)}
						>
							<FloatingImages
								topFloatingImage={secondaryImages[0]}
								bottomFloatingImage={secondaryImages[1]}
								mainImage={mainImage}
								altText={title}
							/>
						</div>
					</div>
				</Box>
			</Link>
		</article >
	);
};

export { Project };
