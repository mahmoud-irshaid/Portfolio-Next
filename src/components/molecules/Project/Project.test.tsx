import { Project } from '../Project';

import { mockProject } from '@mockdata';
import { render } from '@testing-library/react';

describe('Project', () => {
	it('renders correctly', () => {
		const { container } = render(<Project {...mockProject} index={0} />);
		expect(container).toMatchSnapshot();
	});
});
