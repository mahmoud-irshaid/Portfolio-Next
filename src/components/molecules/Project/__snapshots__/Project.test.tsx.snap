// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
<div>
  <article
    class="mt-24 transition first-of-type:mt-0 md:hover:scale-[1.01] lg:w-4/5"
  >
    <a
      class="group"
      href="/project/holland-and-barrett-subscriptions-service"
    >
      <div
        class="glass-light dark:glass-dark mb-6 rounded-xl px-8 py-6 drop-shadow-lg"
      >
        <div
          class="flex flex-col gap-4 md:flex-row"
        >
          <div
            class="flex h-full flex-col md:w-1/2"
          >
            <h1
              class="mb-2 text-xl font-bold md:text-3xl"
            >
              Subscriptions Service
            </h1>
            <div
              class="mb-4 flex items-center gap-2"
            >
              <img
                alt="Holland & Barrett"
                class="rounded-md"
                data-nimg="1"
                decoding="async"
                height="32"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FCzdmdudMQ4mUl4OQJQS2&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FCzdmdudMQ4mUl4OQJQS2&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FCzdmdudMQ4mUl4OQJQS2&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
              <h2
                class="font-bold"
              >
                Holland & Barrett
              </h2>
            </div>
            <p
              class="max-w-xs text-sm"
            >
              <strong>
                Tech used: 
              </strong>
              React, Express, Next.js, Rest APIs, TypeScript, Redux, Node.js, Jest, Cypress
            </p>
          </div>
          <div
            class="-mx-10 -mb-12 md:mx-0 md:-mb-12 md:-mt-8 md:-mr-20"
          >
            <div
              class="relative inline-block w-auto"
            >
              <img
                alt="Subscriptions Service"
                data-nimg="1"
                decoding="async"
                height="275"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2F1WO5SPTlRd2KnNrJH0hF&w=1080&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2F1WO5SPTlRd2KnNrJH0hF&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2F1WO5SPTlRd2KnNrJH0hF&w=1080&q=75 2x"
                style="color: transparent;"
                width="475"
              />
              <div
                class="animate-hover absolute left-12 top-0 z-30 w-1/2 md:left-20"
              >
                <img
                  alt="Subscriptions Service"
                  class="z-30"
                  data-nimg="1"
                  decoding="async"
                  height="210"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FA925zBXITvGXd25ICydM&w=750&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FA925zBXITvGXd25ICydM&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FA925zBXITvGXd25ICydM&w=750&q=75 2x"
                  style="color: transparent;"
                  width="330"
                />
              </div>
              <div
                class="animate-hover animation-delay absolute bottom-0 right-12 z-30 w-1/2 md:right-20"
              >
                <img
                  alt="Subscriptions Service"
                  data-nimg="1"
                  decoding="async"
                  height="210"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FsVY8ZHnBSHiZeeVjFS3f&w=750&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FsVY8ZHnBSHiZeeVjFS3f&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphcms.com%2FsVY8ZHnBSHiZeeVjFS3f&w=750&q=75 2x"
                  style="color: transparent;"
                  width="330"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
  </article>
</div>
`;
